pkg/metamorph/client.go

```diff
@@ -232,22 +246,54 @@ func (m *Metamorph) SubmitTransaction(ctx context.Context, tx *sdkTx.Transaction
 	var response *metamorph_api.TransactionStatus
 	var err error
 	// in case of error try PutTransaction until timeout expires
-	start := time.Now()
-	const interval = 300 * time.Millisecond
-	maxTimeout := time.Duration(5 * time.Second)
-	maxTimeout = max(time.Duration(request.MaxTimeout)*time.Second, maxTimeout)
+
+	maxTimeout := max(time.Duration(request.MaxTimeout)*time.Second, m.maxTimeout)
+
+	retryTicker := time.NewTicker(retryInterval)
+
+	timeoutTimer := time.NewTimer(maxTimeout)
+
```

Comment:
Move this after the first  `m.client.PutTransaction(ctx, request)` call. Add `defer retryTicker.Stop()` to prevent leaks. (I'm not sure if `timeoutTimer` also needs to be stopped manually.)